jobs/integration/test_opa.py

```diff
@@ -0,0 +1,408 @@
+        cmd = f"get ns {name} -o jsonpath='{{.status.phase}}'"
+        log("Waiting for namespace to become active")
+        while result := await kubectl(self.model, cmd):
+            if result.output == "Active":
+                break
+            await asyncio.sleep(1)
+
+        # Delete the namespace
+        cmd = f"delete ns {name}"
+        await kubectl(self.model, cmd)
+
+    async def validate_create_ns_fail(self, name=None):
+        if not name:
+            name = f"test-ns-{random.randint(1, 99999)}"
+
+        cmd = f"create ns {name}"
+        # Creating the namespace should raise
+        log("Creating namespace without any labels")
+        try:
+            await kubectl(self.model, cmd)
+        except JujuRunError as e:
+            err = e.output
+            if err.startswith(
+                "Error from server (InternalError): Internal error occurred"
+            ):
+                # Try again as it might be transient/initialization related
+                await asyncio.sleep(7)
+                try:
+                    await kubectl(self.model, cmd)
+                except JujuRunError as e:
+                    err = e.output
+                else:
+                    pytest.fail("Creating the namespace should fail, but it didn't")
+        else:
+            pytest.fail("Creating the namespace should fail, but it didn't")
+        assert err.startswith(
+            "Error from server (Forbidden): admission webhook "
+            '"validation.gatekeeper.sh" denied the request:'
+        )
+
+    async def deploy_example_policy(self):
+        remote_policy_path = "/tmp/policy.yaml"
+        remote_policy_spec_path = "/tmp/policy-spec.yaml"
+        await scp_to(
+            templates / "gatekeeper-policy.yaml",
+            self.control_plane_unit,
+            remote_policy_path,
+            self.tools.controller_name,
+            self.tools.connection,
+        )
+        await scp_to(
+            templates / "gatekeeper-policy-spec.yaml",
+            self.control_plane_unit,
+            remote_policy_spec_path,
+            self.tools.controller_name,
+            self.tools.connection,
+        )
+        await kubectl(self.model, f"apply -f {remote_policy_path}")
+        await kubectl(self.model, f"apply -f {remote_policy_spec_path}")
+
+    async def destroy_example_policy(self):
+        remote_path = "/tmp/policy.yaml"
+        await scp_to(
+            templates / "gatekeeper-policy.yaml",
+            self.control_plane_unit,
+            remote_path,
+            self.tools.controller_name,
+            self.tools.connection,
+        )
+        log("Delete example policy")
+        await kubectl(self.model, f"delete -f {remote_path}")
+
+    @pytest.fixture(scope="module")
+    async def storage_pool(self, model, k8s_model, tools):
+        # This assumes that we are running on vsphere, should it be
+        # extended to run on other clouds as well?
```

Comment:
@kwmonroe As far as I know, we test on both vSphere and AWS. Is this something that needs to be addressed here?

Reply:
As discussed, I added the `vsphere` mark, should be fine now